.github/workflows/main.yml

```diff
@@ -14,7 +14,7 @@ on:
 jobs:
   build:
     name: JDK ${{ matrix.java_version }}
-    runs-on: macOS-latest
+    runs-on: macos-13
```

Comment:
It's a bit tricky to balance between the 3 versions:
1. emulator api level (currently at `21`, bumping to newer ones means we need to move the abi away from `x86`)
2. emulator abi (due to this being on macos, the only real long term option is `arm64-v8a`)
3. osx version (VMs on Apple silicon is allegedly having some problem with emulators, which [may get resolved in osx15](https://forum.parallels.com/threads/macos-15-sequoia-nested-virtualization-for-m3-macs.364397/), but there doesn't seem to be enough `macos-15`s available at the time of this writing)